### PR TITLE
Genre index remove duplicates

### PR DESCRIPTION
**Pull request by JosephFarina**
in GenreController: to get rid of duplicate genre listings there is a second ArrayList<String> that loops through each artist and checks if their genre is already in the list and adds it if it's not.

I feel like this is probably not the most efficient way of doing it and imagine it would be slow if there were more than 5 artists?

@nickfarina

### CODE REVIEW

File: app/views/genresHome.scala.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@(genres : List[Artist])
+@(artist : List[String])
```
Comment (nickfarina) [inline app/views/genresHome.scala.html]: why did you rename this to artist on a genre page

File: app/views/genreUniquePage.scala.html [hunk 2]
```diff
@@ -9,11 +9,11 @@
 
 <body>
 
-<h1>@genres.get(0).genre</h1>
+<h1>@artist.get(0).genre</h1>
```
Comment (nickfarina) [inline app/views/genreUniquePage.scala.html]: what is this page and why are picking the first artist in the list? - you should move the logic to select a specific artist to the controller, views should do as little logic as possible

File: app/controllers/GenreController.java [hunk 3]
```diff
@@ -14,7 +15,17 @@
...
+        List<Artist> artists = Artist.find.all();
+        List<String> genres = new ArrayList<String>();
+
+        for(Artist artist : artists) {
+            // the genre is already in the list -- add it if it isn't
+            if (!genres.contains(artist.genre)) {
+                genres.add(artist.genre);
+            }
+        }
+
```
Comment (nickfarina) [inline app/controllers/GenreController.java]: This would work fine but its not ideal. There are a few other ways you can do it - 

1) (not ideal) Look into Java's `stream()` methods. They'll let you get a unique list of genres from your `artists` list a bit easier.

2) (better) Figure out how to ask the database directly for all of the unique genres that are in the artist table.

**Comment by nickfarina** [comment]
left comments, don't merge this one until i look at it again

**Comment by JosephFarina** [comment]
Fixed all comments.

File: app/models/Artist.java [hunk 4]
```diff
@@ -20,7 +20,8 @@ public Artist(String name, String genre) {
         this.genre = genre;
     }
 
-    public static Model.Finder<Long, Artist> find = new Model.Finder<Long, Artist>(Long.class, Artist.class);
+    public static Model.Finder<Long, Artist> find
+            = new Model.Finder<Long, Artist>(Long.class, Artist.class);
```
Comment (nickfarina) [inline app/models/Artist.java]: why is there a new line

File: app/controllers/GenreController.java [hunk 5]
```diff
@@ -4,24 +4,28 @@
...
+import com.avaje.*;
 
 public class GenreController extends Controller {
 
     public Result index() {
-        List<Artist> genres = Artist.find.all();
-        return ok(genresHome.render(genres));
+
+        List<Artist> genresEbean = Artist.find.select("genre").setDistinct(true).orderBy("genre asc").findList();
+        return ok(genresHome.render(genresEbean));
```
Comment (nickfarina) [inline app/controllers/GenreController.java]: i like to format long chains like this -

``` java
Artist
  .find
  .select("genre")
  .setDistinct(true)
  .orderBy("genre asc")
.findList();
```

File: app/views/artists.scala.html [hunk 6]
```diff
@@ -10,11 +10,11 @@
 <h1>Artists</h1>
 
 <ul>
-    @for(artist <- artists) {
+    @for(artist <- artist) {
     <li>
         <a href="@routes.ArtistController.show(artist.id)">@artist.name</a>
     </li>
-}
+    }
```
Comment (nickfarina) [inline app/views/artists.scala.html]: these are spaces and not tabs

**Comment by nickfarina** [comment]
are you using sublime text for this? add these to your settings if you are
``` json
{
  "auto_indent": true,
  "caret_style": "phase",
  "color_scheme": "Packages/User/SublimeLinter/dimmed-monokai-no-contrast (SL).tmTheme",
  "ensure_newline_at_eof_on_save": true,
  "fade_fold_buttons": false,
  "file_exclude_patterns":
  [
    "*.a",
    "*.cache",
    "*.class",
    "*.crt",
    "*.db",
    "*.dll",
    "*.dylib",
    "*.exe",
    "*.idb",
    "*.key",
    "*.lib",
    "*.log",
    "*.ncb",
    "*.o",
    "*.obj",
    "*.pdb",
    "*.psd",
    "*.pyc",
    "*.pyo",
    "*.rdb",
    "*.sdf",
    "*.so",
    "*.sock",
    "*.suo",
    "*.woff",
    ".DS_Store"
  ],
  "font_face": "Monaco",
  "font_size": 16,
  "highlight_line": true,
  "highlight_modified_tabs": true,
  "save_on_focus_lost": true,
  "scroll_past_end": true,
  "show_full_path": true,
  "trim_trailing_white_space_on_save": true
}
```

File: app/views/artists.scala.html (hunk 6 above)
Comment (JosephFarina) [inline app/views/artists.scala.html]: @nickfarina Tried to fix it but intelliJ auto converts tabs into 4 spaces in html for some reason. will figure out settings later
